src/conf_mode/protocols_isis.py

```diff
@@ -85,6 +85,24 @@ def verify(isis):
                 if proc_level and proc_level != 'level_1_2' and proc_level != redistribute_level:
                     raise ConfigError('\"protocols isis {0} redistribute ipv4 {2} {3}\" cannot be used with \"protocols isis {0} level {1}\"'.format(process, proc_level, proto, redistribute_level))
 
+        # Segment routing checks
+        if 'segment_routing' in isis_config:
```

Comment:
Please try to make use of `dict_search()` function from `vyos.utils`

Reply:
Will make said changes.